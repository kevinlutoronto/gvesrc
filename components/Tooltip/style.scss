@import '../../styles/base';

.wrapper {
  .content {
    position: relative;
    display: flex;
    pointer-events: none;
    visibility: hidden;
    position: absolute;
    opacity: 0;
    background: white;
    border: 1px solid $cBorder;
    box-shadow: rgba(0,0,0,0.05) 0 2px 5px;
    padding: $space20;
    transform: scale(0.8) !important;
    transform-origin: center top;
    -webkit-transition: all 0.2s ease-in-out;
    transition: all 0.2s ease-in-out;
    &:before {
      pointer-events: none;
      position: absolute;
      content: '';
      font-size: 14px;
      width: 8px;
      height: 8px;
      background: white;
      transform: rotate(45deg);
      z-index: 100;
      top: auto;
      right: auto;
      bottom: 100%;
      left: 50%;
      margin-left: -4px;
      margin-bottom: -4px;
      box-shadow: -1px -1px 0 0 $cBorder;
    }
  }
  &:hover {
    .content {
      visibility: visible;
      pointer-events: auto;
      transform: scale(1) !important;
      opacity: 1;
    }
  }
}