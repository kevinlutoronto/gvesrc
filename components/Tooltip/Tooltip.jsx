// @flow
import React from 'react';
import style from './style.scss';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';

type Props = {|
  children: React$Node,
  content: React$Node,
  data?: Data,
|};

class Tooltip extends React.Component<Props> {
  static defaultProps = {
    children: 'test',
    content: 'Protip: Tooltips can be used to reveal information.',
  };

  render() {
    return (
      <div {...getDataAttrs(this.props.data)} className={style.wrapper}>
        {this.props.children}
        <div className={style.content}>{this.props.content}</div>
      </div>
    );
  }
}

Tooltip.displayName = 'Plasma@Tooltip';

export default Tooltip;
