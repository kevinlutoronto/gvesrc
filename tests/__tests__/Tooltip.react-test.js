/* eslint-disable */

import React from 'react';
import Tooltip from '../../components/Tooltip/Tooltip';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Tooltip Component', () => {
  test('render', () => {
    const component = renderer.create(<Tooltip />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
