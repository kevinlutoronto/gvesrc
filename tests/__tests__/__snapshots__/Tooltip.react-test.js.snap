// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Component render 1`] = `
<div
  className="wrapper"
>
  test
  <div
    className="content"
  >
    Protip: Tooltips can be used to reveal information.
  </div>
</div>
`;
